### ISSUE TITLE
Dependency on uroot

### ISSUE THREAD

**Issue by xbsd** [robjhyndman/forecast#710]
This is more of a suggestion. We use forecast frequently and have found it to be very reliable relative to other options we have explored. Recently whilst working on an AWS Instance with Tensorflow and some other GPU libraries, we noticed that under certain situations, forecast stopped working. It turned out, after many hours of investigation that there were some NVIDIA CUDA related conflicts pertaining to uroot. Since forecast depends on uroot, not only did uroot stop working, but forecast did as well.

In other words, the dependency on CUDA caused the whole app to stop working. Given CUDA can be _extremely_ sensitive in terms of proper versioning and other stringent requirements, it would be great if there were a way to avoid the dependency (or a separate non-CUDA dependent method of installing forecast without uroot).

Just a general suggestion for consideration. Thanks!

**Comment by mitchelloharawild** [comment]
The uroot namespace is loaded when forecast 8.3+ is loaded, which i imagine is causing the issues.
As the unit root tests provided by uroot (ch.test and hegy.test) are not the defaults for nsdiffs, I'm happy to make this package a suggest instead of an import (and test for package install in the nsdiffs function instead).

I think this is a suitable work-around for this issue.

**Comment by robjhyndman** [comment]
Sounds like a reasonable solution.

**Comment by xbsd** [comment]
Thanks! Also, it seems that uroot was last updated on January, 2017. And, as long as it has any dependency on nvcc, it would presumably require more frequent updates/management than usual. So, if any of the functionality can be written without this dependency, it may be better in the long term. We had a recent issue where the entire server had to be rebuilt even after the _experts_ couldn't resolve complex CUDA + kernel errors. Thanks a lot for addressing the matter!

**Comment by mitchelloharawild** [comment]
Hopefully this is better for you now. You should be able to load the forecast package without loading in the uroot package now.
